app/assets/javascripts/settings.coffee

```diff
@@ -1,3 +1,41 @@
 # Place all the behaviors and hooks related to the matching controller here.
 # All this logic will automatically be available in application.js.
 # You can use CoffeeScript in this file: http://coffeescript.org/
+
+# sample code taken (and modified) from:
+# http://rors.org/2012/07/18/custom-confirmation-in-rails
+
+$.rails.allowAction = (link) ->
```

Comment:
Great that you tried this route for the confirmation error, but rails gives you the `data['confirm']` tag to do this on the button itself.

something like, `<%= button_to "Regenerate Key", regen_path, data: { confirm: "Are you sure?" } %>` in the settings/index.html.erb would work as well